### ISSUE TITLE
Scales by make_bdv seem to 'move' the data

### ISSUE THREAD

**Issue by jhennies** [constantinpape/pybdv#32]
Hi @constantinpape,

This code snipped produces a 128^3 dataset of zeros with a 64^3 region of ones in the center (float64) and transforms it into bdv using make_bdv:

```
import numpy as np
from pybdv import make_bdv
from shutil import rmtree
from pybdv.util import open_file, get_key
from matplotlib import pyplot as plt

DOWNSCALE_MODE = 'interpolate'

full_shp = (128,) * 3
pos = (32,) * 3
shp = (64,) * 3
scale_factors = [
    [2, 2, 2],
    [2, 2, 2],
    [2, 2, 2]
]

data = np.zeros(full_shp)
data[
    pos[0]: pos[0] + shp[0],
    pos[1]: pos[1] + shp[1],
    pos[2]: pos[2] + shp[2]
] = 1

out_path = './tmp.n5'
make_bdv(data, out_path, downscale_factors=scale_factors, downscale_mode=DOWNSCALE_MODE)

fig, ax = plt.subplots(nrows=1, ncols=4)
for scale in range(4):
    sl = int(32 / (scale + 1))
    with open_file(out_path, mode='r') as f:
        data = f[get_key(False, timepoint=0, setup_id=0, scale=scale)][sl, :]
    ax[scale].imshow(data)

plt.show()

rmtree(out_path)
```

The outcome of plotting the central slice of each resolution layer is this:

![image](https://user-images.githubusercontent.com/6114948/99582463-6f34a180-29e2-11eb-8c7b-53d734148e3b.png)

It appears to me that each resolution layer is shifted by half a pixel.

Also: setting `DOWNSCALE_MODE = 'nearest'` yields:

![image](https://user-images.githubusercontent.com/6114948/99583312-993a9380-29e3-11eb-95aa-2b636d35d557.png)

Since in this example the data should scale without subpixel locations, I would expect the central (yellow) cube in all four resolutions to appear exactly the same size in the plots - just different scale on the axes. 
Also nearest and interpolate should here yield the same, right?

I don't mind looking into this, maybe you can point me to where the scaling happens?

Best,
Julian

**Comment by constantinpape** [comment]
I think I just ran into the same issue while implementing the in memory dowscaling.
I think it should be fixed once I merge into master, I will let you know when to retest it.

> Also nearest and interpolate should here yield the same, right?

No, `nearest` can only take a given pixel value while `interpolate` can interpolate between multiple values, so we don't expect them to yield the same on the border of the square.

**Comment by jhennies** [comment]
I meant nearest and interpolate to be the same only in this very specific case, where we the boundaries of the square actually coincide with pixel boundaries in all resolution layers. But that's not that important here, I guess ;)

Thanks for looking into it!

**Comment by constantinpape** [comment]
> I meant nearest and interpolate to be the same only in this very specific case, where we the boundaries of the square actually coincide with pixel boundaries in all resolution layers. But that's not that important here, I guess ;)

That's at the root of the issue: in order to do proper interpolation I have a halo to not see this boundary artifacts, but something with this halo computation is going wrong ...

**Comment by jhennies** [comment]
> That's at the root of the issue: in order to do proper interpolation I have a halo to not see this boundary artifacts, but something with this halo computation is going wrong ...

I have a slight feeling, I am also going to run into this kind of issue with the writing of the subset... Currently I am not putting a halo if I have this precise location on the pixel boundaries

**Comment by constantinpape** [comment]
I have merged #31 now and this should fix the issue (but I haven't exactly tested the code here).
